src/classes/OPP_StageMappingUtil.cls

```diff
@@ -93,8 +96,118 @@ public with sharing class OPP_StageMappingUtil {
     @testVisible private static void initializeStates() {
         // These are temporary values to be used only during interim for internal development on Abacus project.
         // This will be updated in March 2019 timeframe with revised code to retrieve from Custom Metadata instead
-        COMMITTED_STATE_SET = new Set<String> {'closed won'};
-        FINALIZED_STATE_SET = new Set<String> {'finalized'};
+        // COMMITTED_STATE_SET = new Set<String> {'closed won'};
+        // FINALIZED_STATE_SET = new Set<String> {'finalized'};
+
+        // If either the committed state set or the finalized state set is null, query for everything and 
+        // reinitialize everything. 
+        if (COMMITTED_STATE_SET == null || FINALIZED_STATE_SET == null) {
+            COMMITTED_STATE_SET = new Set<String>();
+            FINALIZED_STATE_SET = new Set<String>();
+
+            List<Opportunity_Stage_To_State_Mapping__mdt> stageToStateRecords = [Select Id, Opportunity_Stage__c, Opportunity_State__c FROM Opportunity_Stage_To_State_Mapping__mdt];
+
+            for (Opportunity_Stage_To_State_Mapping__mdt mappingRecord : stageToStateRecords) {
+                String oppStage = mappingRecord.Opportunity_Stage__c.trim().toLowerCase();
+
+                if (mappingRecord.Opportunity_State__c.trim().toLowerCase() == 'committed') {
+                    COMMITTED_STATE_SET.add(oppStage);
+                } else {
+                    FINALIZED_STATE_SET.add(oppStage);
+                }
+            }
+        }
+    }
+
+    public static void createFinalizedOppStageToStateRecord() {
+        npe01__Contacts_And_Orgs_Settings__c contactAndOrgCustomSetting = UTIL_CustomSEttingsFacade.getOrgContactsSettings();
+
+        // Only create the finalized opp stage to state record if there's no existing finalized record. 
+        Integer finalizedCount = [Select Count() From Opportunity_Stage_To_State_Mapping__mdt Where Opportunity_State__c =: FINALIZED_STATE_VAL];
+        
+        // Take the value, if there is one, and create a finalized Opp Stage to State custom metadata type record. 
+        if (finalizedCount == 0 && 
+            (contactAndOrgCustomSetting.Payments_Auto_Close_Stage_Name__c != null && 
+            contactAndOrgCustomSetting.Payments_Auto_Close_Stage_Name__c != '')) {
+            
+            String mdTypeName = UTIL_Namespace.StrTokenNSPrefix(OPP_STAGE_TO_STATE_METADATA_NAME);
+            String recordName = FINALIZED_STATE_VAL;
```

Comment:
Recommend Making this FINALIZED_STATE_VAL + '-' + <<Opportunity StageName>>, or 
FINALIZED_STATE_VAL + '-' + now.getTime() + '-' + <<RandomVal>>)

Probably the second one, because we will need to prevent dupes.  Lastly, I would do two things: 

1) Keep this method, but have it call a general method passing a Map of Stage to State Names
2)The general method should make sure it deletes any other records for that Opportunity Stage

Keep in mind you will need additional (method(s)) that are more general when you do the next work item to let them edit these via the UI.

Reply:
I would recommend a clear all, a delete by Stage, and an addAllIn as methods